src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1408,99 +1409,177 @@ static constexpr bool _isHexNumber(const wchar_t wch) noexcept
-//   The only supported spec currently is the following:
-//      spec: a color in the following format:
+//   The supported specs currently are the following:
+//      spec1: a color in the following format:
 //          "rgb:<red>/<green>/<blue>"
-//          where <color> is one or two hex digits, upper or lower case.
+//      spec2: a color in the following format:
+//          "#<red><green><blue>"
+//
+//   In both specs, <color> is a value contains up to 4 hex digits, upper or lower case.
 // Arguments:
 // - string - The string containing the color spec string to parse.
 // - rgb - receives the color that we parsed
 // Return Value:
 // - True if a color was successfully parsed
 bool OutputStateMachineEngine::s_ParseColorSpec(const std::wstring_view string,
                                                 DWORD& rgb) noexcept
+try
 {
-    bool foundRGB = false;
+    bool foundRGBColorSpec = false;
     bool foundValidColorSpec = false;
+
+    bool isSharpSignFormat = false;
+    size_t rgbHexDigitCount = 0;
     std::array<unsigned int, 3> colorValues = { 0 };
+    std::array<unsigned int, 3> parameterValues = { 0 };
     bool success = false;
-    // We can have anywhere between [11,15] characters
-    // 9 "rgb:h/h/h"
-    // 12 "rgb:hh/hh/hh"
-    // Any fewer cannot be valid, and any more will be too many.
-    // Return early in this case.
-    //      We'll still have to bounds check when parsing the hh/hh/hh values
-    if (string.size() < 9 || string.size() > 12)
+    const auto stringSize = string.size();
+
+    // First we look for "rgb:"
+    // Other colorspaces are theoretically possible, but we don't support them.
+    auto curr = string.cbegin();
+    if (stringSize > 4)
     {
-        return false;
+        auto prefix = std::wstring(string.substr(0, 4));
+        std::transform(prefix.begin(), prefix.end(), prefix.begin(), ::towlower);
+        if (prefix.compare(L"rgb:") == 0)
+        {
+            // If all the components have the same digit count, we can have one of the following formats:
+            // 9 "rgb:h/h/h"
+            // 12 "rgb:hh/hh/hh"
+            // 15 "rgb:hhh/hhh/hhh"
+            // 18 "rgb:hhhh/hhhh/hhhh"
+            // Note that the component sizes aren't required to be the same.
+            // Anything in between is also valid, e.g. "rgb:h/hh/h" and "rgb:h/hh/hhh".
+            // Any fewer cannot be valid, and any more will be too many. Return early in this case.
+            if (stringSize < 9 || stringSize > 18)
+            {
+                return false;
+            }
+
+            foundRGBColorSpec = true;
+
+            std::advance(curr, 4);
+        }
     }
 
-    // Now we look for "rgb:"
-    // Other colorspaces are theoretically possible, but we don't support them.
+    // Try the sharp sign format.
+    if (!foundRGBColorSpec && stringSize > 1)
+    {
+        const auto prefix = string.substr(0, 1);
+        if (prefix.compare(L"#") == 0)

```

Comment:
since we validated the size, you can simplify this to...

```suggestion
        if (til::at(prefix, 0) == L'#')
```